x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/plan/logical/InlineStats.java

```diff
@@ -54,7 +117,10 @@ public boolean expressionsResolved() {
 
     @Override
     public List<Attribute> output() {
-        return Expressions.asAttributes(aggregates);
+        if (this.output == null) {
+            this.output = mergeOutputAttributes(Expressions.asAttributes(aggregates), child().output());
```

Comment:
I've updated it now. I can't make a test for it that passes, but I've made one that doesn't pass and added an `Ignore` and stuck it in the follow-up. We need the "expressions in the right hand side" stuff first.

Reply:
Thanks! This will matter once we move the phasing past the logical optimization (instead of after analysis); currently, `Inlinestats.output()` should be dead code.